qadence/pasqal_cloud_connection.py

```diff
@@ -0,0 +1,235 @@
+            )
+        self._validate_parameter_values()
+
+    def _validate_parameter_values(self) -> None:
+        if self.parameter_values is None:
+            return
+        parameter_sizes = [value.size() for value in list(self.parameter_values.values())]
+        if all(len(size) > 1 for size in parameter_sizes):
+            raise ValueError("The dimension of the parameters in parameter_values must be 0 or 1")
+        parameter_lengths = [size[0] for size in parameter_sizes if len(size) == 1]
+        if not all(item == parameter_lengths[0] for item in parameter_lengths):
+            raise ValueError("All parameter values that are arrays, should have the same length")
+
+
+def get_spec_from_model(
+    model: QuantumModel,
+    result_types: list[ResultType],
+    parameter_values: dict[str, Tensor] | None = None,
+    observable: AbstractBlock | None = None,
+) -> WorkloadSpec:
+    """Creates a `WorkloadSpec` from a quantum model.
+
+    This function creates a `WorkloadSpec` from a `QuantumModel` and the other arguments provided.
+    The circuit, that is extracted from the model, is the original circuit that was used to
+    initialize the model, not the backend converted circuit in `model.circuit`. The backend set in
+    the model will be used in the workload specification.
+
+    It is important to note that in case there is an observable defined in the model, it is ignored
+    in the workload specification. To provide an observable to the workload specification, it is
+    only possible to set it in the observable argument of this function.
+
+    Args:
+        model: The quantum model that defines the circuit and backend for the workload spec.
+        result_types: A list of result types that is requested in this workload.
+        parameter_values: The parameter values that should be used during execution of the
+            workload.
+        observable: Observable that is used when `result_types` contains `ResultType.EXPECTATION`.
+            The observable field is mandatory in this case. If not, the value of this field will
+            be ignored. Only a single observable can be passed for cloud submission; providing a
+            list of observables is not supported.
+
+    Returns:
+        A `WorkloadSpec` instance based on the quantum model passed to this function.
+    """
+    circuit = model._circuit.original
+    backend = model._backend_name
+    return WorkloadSpec(circuit, backend, result_types, parameter_values, observable)
+
+
+@dataclass(frozen=True)
+class WorkloadSpecJSON:
+    backend_type: str
+    config: dict[str, Any]
+    workload_type = "qadence_circuit"
+
+
+def _parameter_values_to_json(parameter_values: dict[str, Tensor] | None) -> str:
+    if parameter_values is None:
+        return json.dumps(dict())
+    return json.dumps({key: value.tolist() for key, value in parameter_values.items()})
+
+
+def workload_spec_to_json(workload: WorkloadSpec) -> WorkloadSpecJSON:
```

Comment:
nit: AFAICT, this is not meant to be called by the user so I would suggest making this function "private" and prepending it with an underscore